.goreleaser.yaml

```diff
@@ -0,0 +1,494 @@
+# This is an example .goreleaser.yml file with some sensible defaults.
+# Make sure to check the documentation at https://goreleaser.com
+
+# The lines below are called `modelines`. See `:help modeline`
+# Feel free to remove those if you don't want/need to use them.
+# yaml-language-server: $schema=https://goreleaser.com/static/schema.json
+# vim: set ts=2 sw=2 tw=0 fo=cnqoj
+
+project_name: bao
+
+before:
+  hooks:
+    # You may remove this if you don't use go modules.
+    - go mod tidy
+    # you may remove this if you don't need go generate
+    - go generate ./...
+    - make bootstrap
+
+builds:
+  - id: builds-linux
+    ldflags:
+      - -s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{ .CommitDate }}
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+    goarch:
+      - amd64
+      - arm
+      - arm64
+      - riscv64
+    goarm:
+      - "6"
+      - "7"
+    ignore:
+      - goos: darwin
+        goarm: "6"
+      - goos: freebsd
+        goarm: "7"
+      - goos: illumos
+        goarch: arm
+      - goos: illumos
+        goarch: arm64
+      - goos: linux
+        goarm: "7"
+      - goos: openbsd
+        goarch: riscv64
+      - goos: openbsd
+        goarm: "7"
+      - goos: solaris
+        goarch: arm
+      - goos: solaris
+        goarch: arm64
+    mod_timestamp: "{{ .CommitTimestamp }}"
+    skip: false
+  - id: builds-other
+    ldflags:
+      - -s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{ .CommitDate }}
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - darwin
+      #- illumos
```

Comment:
`illumos` is commented out, but @siepkes had asked for it. :-)

Reply:
Space constraints on the runner. I love to get more in but the runner runs out of disk space currently.

Reply:
Made a self-hosted runner. With the additional headroom I was able to add in Illumos, Solaris, BSD, and Arch Linux support.